From d9a4c0bcfe9c6ae0a861ba66663c2262a035ec48 Mon Sep 17 00:00:00 2001
From: Istio Automation <istio-testing-bot@google.com>
Date: Mon, 10 Jan 2022 08:55:42 -0800
Subject: [release-1.12] Add host header into ZipkinConfig for jaeger when
 using telemetry api  (#36721)

* Add collector_hostname at zipkinConfigGen

* update test code

* Fix istio-testing for run "make gen"

Co-authored-by: IM CHAECHEOL <dlacocjf32@gmail.com>
Co-authored-by: micalgenus <micalgenus@gmail.com>
---
 pilot/pkg/networking/core/v1alpha3/tracing.go      | 2 ++
 pilot/pkg/networking/core/v1alpha3/tracing_test.go | 2 ++
 2 files changed, 4 insertions(+)

diff --git a/pilot/pkg/networking/core/v1alpha3/tracing.go b/pilot/pkg/networking/core/v1alpha3/tracing.go
index d365f0b138..69f095437c 100644
--- a/pilot/pkg/networking/core/v1alpha3/tracing.go
+++ b/pilot/pkg/networking/core/v1alpha3/tracing.go
@@ -235,10 +235,12 @@ func(clusterName string) (*anypb.Any, error) {
 type typedConfigGenFromClusterFn func(clusterName string) (*anypb.Any, error)
 
 func zipkinConfigGen(cluster string) (*anypb.Any, error) {
+	_, _, hostname, _ := model.ParseSubsetKey(cluster)
 	zc := &tracingcfg.ZipkinConfig{
 		CollectorCluster:         cluster,
 		CollectorEndpoint:        "/api/v2/spans",                   // envoy deprecated v1 support
 		CollectorEndpointVersion: tracingcfg.ZipkinConfig_HTTP_JSON, // use v2 JSON for now
+		CollectorHostname:        string(hostname),                  // http host header
 		TraceId_128Bit:           true,
 		SharedSpanContext:        wrapperspb.Bool(false),
 	}
diff --git a/pilot/pkg/networking/core/v1alpha3/tracing_test.go b/pilot/pkg/networking/core/v1alpha3/tracing_test.go
index ed3a9e528e..90e6ce31a0 100644
--- a/pilot/pkg/networking/core/v1alpha3/tracing_test.go
+++ b/pilot/pkg/networking/core/v1alpha3/tracing_test.go
@@ -367,10 +367,12 @@ func fakeTracingConfig(provider *tracingcfg.Tracing_Http, randomSampling float64
 }
 
 func fakeZipkinProvider(expectClusterName, expectProviderName string) *tracingcfg.Tracing_Http {
+	_, _, hostname, _ := model.ParseSubsetKey(expectClusterName)
 	fakeZipkinProviderConfig := &tracingcfg.ZipkinConfig{
 		CollectorCluster:         expectClusterName,
 		CollectorEndpoint:        "/api/v2/spans",
 		CollectorEndpointVersion: tracingcfg.ZipkinConfig_HTTP_JSON,
+		CollectorHostname:        string(hostname),
 		TraceId_128Bit:           true,
 		SharedSpanContext:        wrapperspb.Bool(false),
 	}
-- 
2.35.3

